luizfiorentino/my-city-app
# Multiform feature

# What this pull request does

## What

- The issue form is transformed into a "multi-step" from instead
- It will be divided into "blocks":
- Step 1: infos - a) userName, b) email (optional) and c) description
- Step 2: location
-  Step 3: pictures(optional), and
- Step 4: summary and confirm data

## Why

- Gathering all the information from the form in a single page can be overwhelming to the user.
- At every step, the user is guided to fill in the info with more clarity, getting instructions (e.g. what info is need, if it is optional or not)
- This format is very suitable with different screen sizes, especially the mobile one

## How

- The original form and its validation  are divided into three smaller forms
- At each step the data is stored in the context with `useContext`
- The context also contains a variable related to the different steps: when the user clicks "Next", the context changes to display the next one, and when he/she clicks "back", it resets the stored data from the previous step and the correspondent form is displayed
- In the last step, the user can confirm the data, and submit it. The info stored with useContext is used in the api POST request to create a new issue

## Steppings to test this

- Go to `/` (issue form) and fill it in, click "Next" and so on
- Try the "Go back" button (you can even console.log the context to confirm the data is cleared, and then new data is filled in)
- Use different screen sizes to see how better is the multi-step form in comparison to the previous model of a single form


## Anything else

Adding a "edit" button in the summary (step 4) so that the user can change the info also from there, with no need to use the "Go back" button. 

https://github.com/luizfiorentino/my-city-app/assets/96445830/8f681a87-7670-4bbe-8175-590fe4cd5dc3
